### ISSUE TITLE
[GEO] False positive match using geo_shape

### ISSUE THREAD

**Issue by nedcampion** [elastic/elasticsearch#10583]
Hi Elasticsearch Team,

I wanted to surface an issue regarding a geo query that seems to be giving me a false positive result. I have [a gist here that can be used to recreate what I am seeing](https://gist.github.com/nedcampion/0bbc30f2a999dc904ebf)

I am testing against 1.5.1. In short I create an index with a `geo_shape` field with the maximum precision, I then index a document containing a polygon that describes the boundary of Mexico and I then query against that index using an envelope that describes a small area in the southern tip of Texas. The query returns the document containing Mexico which is a false positive match.

I realize that [geo_shape does not provide 100% accuracy depending on how it is configured](http://www.elastic.co/guide/en/elasticsearch/reference/1.3/mapping-geo-shape-type.html#_accuracy) but the region I am querying with seems many miles away from the Mexico border and the precision is set to 1m. Is this a bug or am I missing something.

**Comment by clintongormley** [comment]
@nknize please could you take a look

**Comment by nknize** [comment]
This is related to #9691 (and similar #9860) a fix will be in place very very shortly.
 
tl;dr:  if you really want full precision you need to lower the distance_error_pct value. Be careful, you can quickly blow out memory.

In the meantime, if you want to get accurate results and preserve the most memory your best bet is to post-filter the results in the application layer (resultShape.relate(candidateShape))
